src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -71,6 +72,20 @@ void setConversationStore(ConversationStore conversationStore) {
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     List<Conversation> conversations = conversationStore.getAllConversations();
+
+    // Filter out test data to non-admins
+    HttpSession session = request.getSession();
+    boolean adminStatus = session.getAttribute("isAdmin") != null ? (boolean) session.getAttribute("isAdmin") : false;
+
+    if (adminStatus == false) {
+      TestDataGenerator tdg = TestDataGenerator.getInstance();
```

Comment:
Hm, this will cause a full TestDataGenerator, including the PersistentStorageAgent, to get created in the ConversationServletTest. It would probably be better practice to set the TestGenerator the same way the datastores are set, so we can mock it out for testing.

Speaking of which, are there tests for this filtering logic?